scripts/contributor_dashboard_debug_test.py

```
@@ -0,0 +1,260 @@
+
+        with self.request_swap, sign_up_swap, begin_session_swap, (
+                init_app_swap) as init_app_counter:
+            self.contributor_dashboard_debug.populate_debug_data()
+
+        self.assertEqual(init_app_counter.times_called, 1)
+        self._assert_user_roles(
+            contributor_dashboard_debug.SUPER_ADMIN_USERNAME,
+            contributor_dashboard_debug.SUPER_ADMIN_ROLES)
+        self._assert_can_submit_question_suggestions(
+            contributor_dashboard_debug.CONTRIBUTOR_USERNAME)
+        self._assert_generate_sample_new_structures_data()
+        self._assert_topics_in_classroom(
+            contributor_dashboard_debug.CLASSROOM_NAME)
+
+    def mock_login_as_admin(self, email: str) -> None:
+        """Sets the environment variables to simulate a login of admin."""
+        self.login(email, is_super_admin=True)
+
+    def test_sign_up_new_user(self) -> None:
+        email = 'user1@example.com'
+        username = 'user1'
+        create_user_swap = self.swap(
+            firebase_auth, 'create_user', self.mock_firebase_auth_create_user)
+        begin_session_swap = self.swap(
+            self.contributor_dashboard_debug, '_sign_in', self.login)
+        load_template_swap = self.swap(
+            base, 'load_template', test_utils.mock_load_template)
+
+        with self.request_swap, create_user_swap, begin_session_swap, (
+                load_template_swap):
+            self.contributor_dashboard_debug._sign_up_new_user(email, username) # pylint: disable=protected-access
+
+        self.assertIsNotNone(self.firebase_sdk_stub.get_user_by_email(email))
+        self.assertEqual(
+            user_services.get_user_settings_from_email(email).username, # type: ignore
+            username)
+
+    def mock_firebase_auth_create_user(self, **kwargs: Any) -> Any:
+        """Mock for firebase_auth.create_user()."""
+        email = kwargs['email']
+        auth_id = self.get_auth_id_from_email(email) # type: ignore
```

I'm a little suspicious of all these `type: ignore` comments. Can we avoid them? If they really are needed, please follow the guidance in https://github.com/oppia/oppia/wiki/Backend-Type-Annotations, specifically:

* Ignore comments should be for a particular error
* For all ignores except `[no-untyped-call]`, add a comment explaining why we are ignoring the error

---

You are right. We can avoid most of them. There are only [no-untyped-call] ignores now.